editor/src/messages/tool/tool_messages/spline_tool.rs

```diff
@@ -534,3 +534,308 @@ fn delete_preview(tool_data: &mut SplineToolData, responses: &mut VecDeque<Messa
+
+		// Evaluate the graph to ensure everything is processed
+		editor.eval_graph().await;
+
+		if let Some(points) = get_spline_points(&mut editor).await {
+			assert!(points.len() >= 3, "Spline should have at least 3 points");
+
+			// Calculating expected points in document coordinates
+			let document = editor.active_document();
+			let viewport_to_document = document.metadata().document_to_viewport.inverse();
+
+			let expected_points = vec![
+				viewport_to_document.transform_point2(DVec2::new(50.0, 50.0)),
+				viewport_to_document.transform_point2(DVec2::new(100.0, 50.0)),
+				viewport_to_document.transform_point2(DVec2::new(150.0, 100.0)),
+			];
+
+			// Check each key point is close to the expected position
+			for (i, (point, expected)) in points.iter().take(3).zip(expected_points.iter()).enumerate() {
+				assert!(
+					(*point - *expected).length() < 1.0,
+					"Point {i} should be close to expected position. Got: {point:?}, expected: {expected:?}"
+				);
+			}
+		} else {
+			panic!("Could not get spline points");
+		}
+	}
+
+	#[tokio::test]
+	async fn test_spline_with_panned_view() {
+		let mut editor = EditorTestUtils::create();
+		editor.new_document().await;
+
+		let pan_amount = DVec2::new(200.0, 150.0);
+		editor.handle_message(NavigationMessage::CanvasPan { delta: pan_amount }).await;
+
+		editor.select_tool(ToolType::Spline).await;
+		editor.left_mousedown(50.0, 50.0, ModifierKeys::empty()).await;
+		editor
+			.mouseup(
+				EditorMouseState {
+					editor_position: DVec2::new(50.0, 50.0),
+					mouse_keys: MouseKeys::empty(),
+					scroll_delta: ScrollDelta::default(),
+				},
+				ModifierKeys::empty(),
+			)
+			.await;
+
+		editor.left_mousedown(100.0, 50.0, ModifierKeys::empty()).await;
+		editor
+			.mouseup(
+				EditorMouseState {
+					editor_position: DVec2::new(100.0, 50.0),
+					mouse_keys: MouseKeys::empty(),
+					scroll_delta: ScrollDelta::default(),
+				},
+				ModifierKeys::empty(),
+			)
+			.await;
+
+		editor.left_mousedown(150.0, 100.0, ModifierKeys::empty()).await;
+		editor
+			.mouseup(
+				EditorMouseState {
+					editor_position: DVec2::new(150.0, 100.0),
+					mouse_keys: MouseKeys::empty(),
+					scroll_delta: ScrollDelta::default(),
+				},
+				ModifierKeys::empty(),
+			)
+			.await;
+
+		editor.handle_message(SplineToolMessage::Confirm).await;
+		editor.eval_graph().await;
+
+		if let Some(points) = get_spline_points(&mut editor).await {
+			assert!(points.len() >= 3, "Spline should have at least 3 points");
+			let document = editor.active_document();
+			let viewport_to_document = document.metadata().document_to_viewport.inverse();
+			let expected_points = vec![
+				viewport_to_document.transform_point2(DVec2::new(50.0, 50.0)),
+				viewport_to_document.transform_point2(DVec2::new(100.0, 50.0)),
+				viewport_to_document.transform_point2(DVec2::new(150.0, 100.0)),
+			];
+			for (i, (point, expected)) in points.iter().take(3).zip(expected_points.iter()).enumerate() {
```

Comment:
There is no need to use `.take()` since zip will end when either of the iterators ends. See: https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.zip